src/app/profile/user.service.ts

```diff
@@ -32,18 +34,23 @@ export class UserService extends BaseResourceService<User> implements OnDestroy
     protected fds: FirestoreDebugService
   ) {
     super(alertService, afs, 'users', fds);
+
+    const sharedFirebaseUser$ = this.authService.firebaseUser$.pipe(shareReplay({ bufferSize: 1, refCount: true }));
+    this.publicUser$ = sharedFirebaseUser$.pipe(
+      switchMap(firebaseUser => this.publicUserService.get(firebaseUser.uid))
+    );
+    this.user$ = sharedFirebaseUser$.pipe(switchMap(firebaseUser => this.get(firebaseUser.uid)));
     // load roles or initialize roles array if public user document does not exist or roles array is not defined
-    this.roles$ = this.publicUserService
-      .get(this.authService.getUserId())
-      .pipe(map(publicUser => (publicUser && publicUser.roles ? publicUser.roles : [])));
+    this.roles$ = this.publicUser$.pipe(map(publicUser => (publicUser && publicUser.roles ? publicUser.roles : [])));
   }
 
   ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
 
+  /** @deprecated use user$ instead */
```

Comment:
getUser wird nur 7 mal aufgerufen, evtl. kannst du diese Calls gleich alle umschreiben und die Methode löschen?

Reply:
I was trying not to get into the refactoring rabbit-hole. But probably makes sense in this case :)